x/authority/keeper/authorization_list.go

```diff
@@ -24,3 +27,48 @@ func (k Keeper) GetAuthorizationList(ctx sdk.Context) (val types.AuthorizationLi
 	k.cdc.MustUnmarshal(b, &val)
 	return val, true
 }
+
+// IsAuthorized checks if the address is authorized for the given policy type
+func (k Keeper) IsAuthorized(ctx sdk.Context, address string, policyType types.PolicyType) bool {
+	policies, found := k.GetPolicies(ctx)
+	if !found {
+		return false
+	}
+	for _, policy := range policies.Items {
+		if policy.Address == address && policy.PolicyType == policyType {
+			return true
+		}
+	}
+	return false
+}
+
+// CheckAuthorization checks if the signer is authorized to sign the message
+// It uses both the authorization list and the policies to check if the signer is authorized
+func (k Keeper) CheckAuthorization(ctx sdk.Context, msg sdk.Msg) error {
```

Comment:
The `CheckSigner` function is similar, the only difference being instead of sending`MsgUrl`  , we are are sending the `PolicyRequired`. 

We can move the the following logic into `CheckSigner` 
```
	policyRequired, err := authorizationsList.GetAuthorizedPolicy(msgURL)
	if err != nil {
		return errors.Wrap(types.ErrAuthorizationNotFound, fmt.Sprintf("msg: %v", msgURL))
	}
	if policyRequired == types.PolicyType_groupEmpty {
		return errors.Wrap(types.ErrInvalidPolicyType, fmt.Sprintf("Empty policy for msg: %v", msgURL))
	}
```
But I don't see a lot of value in that , we would make the `CheckAuthorization` tests slightly cleaner while sacrificing the `CheckSigner` tests .

Reply:
ok